pkg/v1/tkg/client/package_helper.go

```diff
@@ -27,28 +28,64 @@ func GetClusterBootstrap(managementClusterClient clusterclient.Client, clusterNa
 	if clusterBootstrap.Spec.Kapp != nil && clusterBootstrap.Spec.Kapp.ValuesFrom != nil && clusterBootstrap.Spec.Kapp.ValuesFrom.ProviderRef != nil {
-		name := strings.TrimSuffix(clusterBootstrap.Spec.Kapp.ValuesFrom.ProviderRef.Name, suffixStr)
-		packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: clusterBootstrap.ObjectMeta.Namespace}})
+		pkg, err := getPackage(regionalClusterClient, clusterBootstrap.Spec.Kapp.RefName, clusterBootstrap.Namespace, clusterName)
+		if err != nil {
+			return packages, err
+		}
+		packages = append(packages, *pkg)
 	}
-	// Core packages CNI, CSI and CPI (other than Kapp) installed in tkg-system namespace in case of tkgm, and in case of tkgs installed in vmware-system-tkg namespace
+
+	// Core packages CNI, CSI and CPI (other than Kapp) installed in workload cluster, in tkg-system namespace in case of tkgm, and in case of tkgs installed in vmware-system-tkg namespace
 	if clusterBootstrap.Spec.CNI != nil && clusterBootstrap.Spec.CNI.ValuesFrom != nil && clusterBootstrap.Spec.CNI.ValuesFrom.ProviderRef != nil {
-		name := strings.TrimSuffix(clusterBootstrap.Spec.CNI.ValuesFrom.ProviderRef.Name, suffixStr)
-		packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: corePackagesNamespace}})
+		pkg, err := getPackage(workloadClusterClient, clusterBootstrap.Spec.CNI.RefName, corePackagesNamespace, clusterName)
+		if err != nil {
+			return packages, err
+		}
+		packages = append(packages, *pkg)
 	}
 	if clusterBootstrap.Spec.CSI != nil && clusterBootstrap.Spec.CSI.ValuesFrom != nil && clusterBootstrap.Spec.CSI.ValuesFrom.ProviderRef != nil {
-		name := strings.TrimSuffix(clusterBootstrap.Spec.CSI.ValuesFrom.ProviderRef.Name, suffixStr)
-		packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: corePackagesNamespace}})
+		pkg, err := getPackage(workloadClusterClient, clusterBootstrap.Spec.CSI.RefName, corePackagesNamespace, clusterName)
+		if err != nil {
+			return packages, err
+		}
+		packages = append(packages, *pkg)
 	}
 	if clusterBootstrap.Spec.CPI != nil && clusterBootstrap.Spec.CPI.ValuesFrom != nil && clusterBootstrap.Spec.CPI.ValuesFrom.ProviderRef != nil {
-		name := strings.TrimSuffix(clusterBootstrap.Spec.CPI.ValuesFrom.ProviderRef.Name, suffixStr)
-		packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: corePackagesNamespace}})
+		pkg, err := getPackage(workloadClusterClient, clusterBootstrap.Spec.CPI.RefName, corePackagesNamespace, clusterName)
+		if err != nil {
+			return packages, err
+		}
+		packages = append(packages, *pkg)
 	}
-	return packages
+	return packages, nil
+}
+
+//getPackage takes clusterclient, package name, namespace and cluster name, queries for package details and give package object (which has package name and namespace details)
```

Comment:
need a space here (after //)? also this is not an exported function, so not sure go docs will pick it up.

Reply:
yeah, not sure why lint didn't report it!!